lua/telescope/make_entry.lua

```diff
@@ -304,33 +302,35 @@ function make_entry.gen_from_buffer(opts)
       display_bufname = entry.filename
     end
 
-    return string.format("%" .. opts.bufnr_width .. "d : %s",
-                         entry.bufnr, display_bufname)
+    return displayer {
+      entry.bufnr,
+      entry.indicator,
+      display_bufname .. ":" .. entry.lnum,
+    }
   end
 
   return function(entry)
-    local bufnr_str = tostring(entry)
-    local bufname = path.normalize(vim.api.nvim_buf_get_name(entry), cwd)
-
+    local bufname = entry.info.name ~= "" and entry.info.name or '[No Name]'
```

Comment:
That fixes #245 and is the reason why we no longer can use bat for previewing buffers, but i can't switch to unnamed buffers. Do you know why? I am getting `Vim(edit):E499: Empty file name for '%' or '#', only works with ":p:h"`.
Do i have to use something like `vim.api.nvim_win_set_buf(current_buffer, unnamed_bufnr)`